graylog2-web-interface/src/views/components/sidebar/NavItem.jsx

```diff
@@ -1,65 +1,130 @@
 // @flow strict
 import * as React from 'react';
 import PropTypes from 'prop-types';
-// $FlowFixMe: could not find types
 import { SizeMe } from 'react-sizeme';
+import styled, { css } from 'styled-components';
 
-import styles from './SideBar.css';
+import { Icon } from 'components/common';
 import CustomPropTypes from '../CustomPropTypes';
 
+type StyleProps = {
+  isSelected: boolean,
+  expandRight: boolean,
+};
+
+export const Title: React.ComponentType<StyleProps> = styled.div`
+  padding: 9px 10px;
+  display: flex;
+  align-items: center;
+  cursor: pointer;
+  position: relative;
+
+  ${props => (props.isSelected ? css`
+    color: #FF3633;
+    background: #393939;
+  ` : '')}
+
+  ${props => (props.isSelected && props.expandRight ? css`
+    &::after {
+      content: ' ';
+      display: block;
+      position: absolute;
+      right: 0;
+      width: 0;
+      height: 0;
+      border-top: 15px solid transparent;
+      border-right: 15px solid white;
+      border-bottom: 15px solid transparent;
+    }
+  ` : '')}
+`;
+
+const TitleText = styled.div`
+  font-size: 16px;
+  display: inline;
+  margin-left: 10px;
+  overflow: hidden;
+  white-space: nowrap;
+`;
+
+const TitleIcon = styled.div`
+  width: 25px;
+  text-align: center;
+  font-size: 20px;
+  cursor: pointer;
+`;
+
+const Content: React.ComponentType<StyleProps> = styled.div`
+  color: #666666;
+  background: #FFFFFF;
+  box-shadow:
+          inset 0px 13px 5px -10px #CCC,
+          inset 0px -13px 5px -10px #CCC;
+
+  ${props => (props.isSelected ? css`
+    padding: 20px;
+  ` : css`
+    max-height: 0;
+  `)}
+
+  ${props => (props.isSelected && props.expandRight ? css`
+    position: absolute !important;
+    top: 0;
+    left: 100%;
+    border: 0;
+    bottom: 0;
+    padding: 20px;
+    width: 300px;
+    overflow-y: hidden;
+  ` : '')}
+`;
+
 type Props = {
   isOpen: boolean,
   isSelected: boolean,
-  isLast: boolean,
+  expandRight: boolean,
   text: string,
-  icon: React.Element<any>,
+  icon: string,
   onClick: (string) => void,
   children: React.Element<any>,
 }
 
-const NavItem = ({ isOpen, isSelected, isLast, text, children, icon, onClick }: Props) => {
-  const selectedColor = isSelected ? styles.selected : '';
+const NavItem = ({ isOpen, isSelected, expandRight, text, children, icon, onClick }: Props) => {
   // eslint-disable-next-line no-nested-ternary
-  const selected = isSelected
-    ? (isLast
-      ? styles.openFieldContent
-      : styles.contentOpen)
-    : styles.contentClosed;
-
   return (
-    <div>
-      <div role="presentation" onClick={onClick} className={`${styles.sidebarNav} ${selectedColor}`}>
-        <div className={styles.sidebarIcon}>{icon}</div>
-        {(isOpen && <div className={styles.sidebarNavFont}>{text}</div>)}
-      </div>
+    <React.Fragment>
+      <Title role="presentation" onClick={onClick} isSelected={isSelected} expandRight={expandRight}>
```

Comment:
@Graylog2/frontend what do we think about adding a lint rule that requires line breaks after `N` props/attributes? Maybe at 3 or 4 we require breaks for ease of readability?

Reply:
+1
I am a fan of this rule.

Reply:
IMO there are two ways to solve this:

- implicitly by the `max-len` rule (which is currently off), enforcing line breaks after N characters
- force switching to a multi-line declaration (with a single prop per line) after 3-4 props

I am open for both options.